test/text_map_codec.js

```diff
@@ -24,8 +24,29 @@ import ConstSampler from '../src/samplers/const_sampler.js';
 import InMemoryReporter from '../src/reporters/in_memory_reporter.js';
 import opentracing from 'opentracing';
 import Tracer from '../src/tracer.js';
+import MetricsContainer from '../src/metrics/metrics.js';
+import LocalMetricFactory from '../src/metrics/local/metric_factory.js';
+import TestUtils from '../src/test_util.js';
 
 describe('Text Map Codec should', () => {
+    it ('report metric when failing to decode tracer stawte', () => {
```

Comment:
don't we already have a unit test for failure to extract context? I'd rather add metrics check there than repeat the boilerplate